ext/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -0,0 +1,65 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Instrument `redis`_ to report Redis queries.
+
+There are two options for instrumenting code. The first option is to use
+the `opentelemetry-auto-instrumentation` executable which will automatically
+patch your Redis client. The second is to programmatically enable
+instrumentation via the following code:
+
+.. _redis: https://pypi.org/project/redis/
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry import trace
+    from opentelemetry.instrumentation.redis import RedisInstrumentor
+    from opentelemetry.sdk.trace import TracerProvider
+    import redis
+
+    trace.set_tracer_provider(TracerProvider())
+
+    # You can patch redis specifically
+    RedisInstrumentor().instrument(tracer_provider=trace.get_tracer_provider())
```

Comment:
I think the general idea is to omit the `tracer_provider` parameter in the basic examples and let it as an advance use case. By default all the integrations should default to  `trace.get_tracer_provider()` when the tracer provider is not passed.